diff --git a/dist/jsQR.js b/dist/jsQR.js
--- a/dist/jsQR.js
+++ b/dist/jsQR.js
@@ -341,7 +341,8 @@ function scan(matrix) {
             return {
                 binaryData: decoded.bytes,
                 data: decoded.text,
                 chunks: decoded.chunks,
+                matrix: extracted.matrix,
                 version: decoded.version,
                 location: {
                     topRightCorner: extracted.mappingFunction(location_1.dimension, 0),
@@ -374,9 +375,12 @@ function jsQR(data, width, height, providedOptions) {
     var result = scan(tryInvertedFirst ? inverted : binarized);
     if (!result && (options.inversionAttempts === "attemptBoth" || options.inversionAttempts === "invertFirst")) {
         result = scan(tryInvertedFirst ? binarized : inverted);
     }
     return result;
 }
+jsQR.binarize = binarizer_1.binarize;
+jsQR.extract = extractor_1.extract;
+jsQR.locate = locator_1.locate;
 jsQR.default = jsQR;
 exports.default = jsQR;
 
